dotcom-rendering/cypress/e2e/parallel-5/liveblog.interactivity.cy.js

```diff
@@ -56,17 +56,17 @@ describe('Liveblogs', function () {
 				html: '<p>New block</p>',
 				mostRecentBlockId: 'abc',
 			});
-			cy.get(`[data-cy="toast"]`).should('exist');
-			cy.contains('1 new update');
-			cy.window().then(function (win) {
-				win.mockLiveUpdate({
-					numNewBlocks: 1,
-					html: '<p>New block</p>',
-					mostRecentBlockId: 'abc',
-				});
-				cy.contains('2 new updates');
+		});
+		cy.get(`[data-cy="toast"]`).should('exist');
+		cy.contains('1 new update');
+		cy.window().then(function (win) {
+			win.mockLiveUpdate({
+				numNewBlocks: 1,
+				html: '<p>New block</p>',
+				mostRecentBlockId: 'abc',
 			});
 		});
+		cy.contains('2 new updates');
```

Comment:
Given [this documentation](https://docs.cypress.io/api/commands/then#Whats-the-difference-between-then-and-shouldand)

> Using .then() allows you to use the yielded subject in a callback function and should be used when you need to manipulate some values or do some actions.
> 
> When using a callback function with .should() or .and(), on the other hand, there is special logic to rerun the callback function until no assertions throw within it.

Should this be chained?

```suggestion
		cy.window().then(function (win) {
			win.mockLiveUpdate({
				numNewBlocks: 1,
				html: '<p>New block</p>',
				mostRecentBlockId: 'abc',
			});
		}).should('contains', '2 new updates');
```

I'm not sure if we'd hit race conditions otherwise?

Reply:
I don't think that `cy.contains` is run until the previous command is complete, so we wouldn't hit a race condition here.

I believe that bit of documentation advises the user not to put assertions within a `.then` as these will not be retried. The mock update is perfect for us to go in to a `.then` as we don't want this code to be retried.

If the mock update is slow and the text doesn't appear on the page immediately, the `cy.contains` will wait the standard 4 seconds for the text to appear. I don't believe it makes a difference whether it is chained on the end of the `cy.window().then(...)` or not.

Reply:
> I believe that bit of documentation advises the user not to put assertions within a .then as these will not be retried.

That's the important bit, let's :shipit: